pkg/builder/cluster/builder.go

```diff
@@ -57,7 +57,19 @@ func (op *operation) Checkpoint(status *v1alpha1.BuildStatus) error {
 	status.StartTime = op.builder.podCreationTime
 	status.StepStates = nil
 	status.StepsCompleted = nil
-	for _, s := range op.statuses {
+
+	// Always ignore the first pod status, which is creds-init.
+	skip := 1
+	if build.Spec.Source != nil {
+		// If the build specifies source, skip another container status, which
+		// is the source-fetching container.
+		skip++
+	}
+	if skip > len(op.statuses) {
```

Comment:
Looks like there is no test coverage for this condition `skip> len(op.statuses)`.